src/System.IO.Pipelines/src/System/IO/Pipelines/ThreadPoolScheduler.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    internal sealed class ThreadPoolScheduler : PipeScheduler
+    {
+        public override void Schedule(Action action)
+        {
+#if NETCOREAPP2_1
```

Comment:
This #if maze is too hard to follow. I think it'd be better to have conditionally imported files named ThreadPoolScheduler.NetCoreApp21.cs, ThreadPoolScheduler.NetStandard2.cs instead.

Reply:
Is it worth it? They don't even intersect, three small completely different versions depending on the target.

Reply:
The lack of commonality is exactly why separate files is preferable to #if's. #if's are our least-preferred mechanism for version-specific coding - mainly to be used when 95% of a file is the same between versions and only a handful of lines need to be tweaked. Here, almost every material line is conditional on something - it's very hard to follow.